core/deployment/src/main/java/io/quarkus/deployment/SnapStartProcessor.java

```diff
@@ -37,7 +37,7 @@ public void processSnapStart(BuildProducer<PreloadClassesEnabledBuildItem> prelo
                 return;
             }
 
-        } else if (defaultVal == null || !defaultVal.isPresent() || !defaultVal.get().isDefaultValue()) {
```

Comment:
this is a quick win and if not done here I would encourage and suggest to run this within an supplemental PR